shipyard_blueprints/shipyard-bp-utils/tests/files_test.py

```diff
@@ -0,0 +1,40 @@
+from shipyard_bp_utils import files
+
+import pytest
+import os
+
+
+@pytest.mark.parametrize(
+    "destination_file_name,file_number,expected_result",
+    [
+        ("output.txt", None, "output.txt"),
+        ("output.txt", 1, "output_1.txt"),
```

Comment:
One thing you need to test for is when the `source_full_path` isn't a nested directory structure, but just a file in the current working directory